core/trino-main/src/main/java/io/trino/connector/InMemoryCatalogStore.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import io.trino.spi.connector.CatalogHandle.CatalogVersion;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static io.trino.spi.connector.CatalogHandle.createRootCatalogHandle;
+
+public class InMemoryCatalogStore
+        implements CatalogStore
+{
+    private final ConcurrentMap<String, CatalogProperties> catalogs = new ConcurrentHashMap<>();
+
+    @Override
+    public Collection<CatalogProperties> getCatalogs()
+    {
+        return ImmutableList.copyOf(catalogs.values());
+    }
+
+    @Override
+    public CatalogProperties createCatalogProperties(String catalogName, String connectorName, Map<String, String> properties)
+    {
+        return new CatalogProperties(
+                createRootCatalogHandle(catalogName, computeCatalogVersion(catalogName, connectorName, properties)),
+                connectorName,
+                properties);
+    }
+
+    @Override
+    public void addCatalog(CatalogProperties catalogProperties)
+    {
+        catalogs.put(catalogProperties.getCatalogHandle().getCatalogName(), catalogProperties);
+    }
+
+    @Override
+    public void removeCatalog(String catalogName)
+    {
+        catalogs.remove(catalogName);
+    }
+
+    private static CatalogVersion computeCatalogVersion(String catalogName, String connectorName, Map<String, String> properties)
```

Comment:
Extract a utility class for this? Or make the one in `FileCatalogStore` package private?